controllers/component_image_controller.go

```diff
@@ -72,8 +72,8 @@ type ComponentReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
 
-	QuayClient       quay.QuayService
-	QuayOrganization string
+	BuildQuayClient      func() quay.QuayService
+	ReadQuayOrganization func() string
```

Comment:
You still read the file two times in a row

Reply:
as we are removing Organization read at runtime, we are going to have a read at startup for the Organization and a read if the client is needed.